src/test/java/org/broadinstitute/hellbender/tools/sv/cluster/SVCollapserTest.java

```diff
@@ -147,410 +301,1091 @@ public void collapseAltAllelesTest(final List<List<Allele>> recordGenotypeAllele
+
+        // Test as genotype attributes
+        final List<Genotype> genotypes = inputAttributesList.stream()
+                .map(m -> new GenotypeBuilder().attributes(m).make())
+                .collect(Collectors.toList());
+        final Map<String, Object> testGenotypeMap = collapser.collapseGenotypeAttributes(genotypes, expectedCopyNumber);
+        Assert.assertEquals(testGenotypeMap, expectedAttributes);
+
+        // Test as variant attributes
+        final List<SVCallRecord> variants = IntStream.range(0, inputAttributesList.size())
+                .mapToObj(i -> SVTestUtils.newNamedDeletionRecordWithAttributes(variantIds.get(i), inputAttributesList.get(i)))
+                .collect(Collectors.toList());
+        final Map<String, Object> expectedAttributesWithMembers = new HashMap<>(expectedAttributes);
+        expectedAttributesWithMembers.put(GATKSVVCFConstants.CLUSTER_MEMBER_IDS_KEY, variantIds);
+        expectedAttributesWithMembers.remove(GATKSVVCFConstants.EXPECTED_COPY_NUMBER_FORMAT);
+        final Map<String, Object> testVariantMap = collapser.collapseVariantAttributes(variants);
+        Assert.assertEquals(testVariantMap, expectedAttributesWithMembers);
+    }
+
+    private Map<String, Object> createGenotypeTestAttributes(final Integer expectedCopyNumber, final Integer copyNumber, final String testVal) {
+        final String[] keys = new String[]{
+                GATKSVVCFConstants.EXPECTED_COPY_NUMBER_FORMAT,
+                GATKSVVCFConstants.COPY_NUMBER_FORMAT,
+                TEST_KEY_1
+        };
+        final Object[] vals = new Object[]{expectedCopyNumber, copyNumber, testVal};
+        return SVTestUtils.keyValueArraysToMap(keys, vals);
+    }
+
+    private Map<String, Object> createGenotypeTestAttributes(final Integer expectedCopyNumber, final String testVal) {
+        final String[] keys = new String[]{
+                GATKSVVCFConstants.EXPECTED_COPY_NUMBER_FORMAT,
+                TEST_KEY_1
+        };
+        final Object[] vals = new Object[]{expectedCopyNumber, testVal};
+        return SVTestUtils.keyValueArraysToMap(keys, vals);
+    }
+
+    private Map<String, Object> createGenotypeTestAttributes(final Integer expectedCopyNumber, final Integer copyNumber) {
+        final String[] keys = new String[]{
+                GATKSVVCFConstants.EXPECTED_COPY_NUMBER_FORMAT,
+                GATKSVVCFConstants.COPY_NUMBER_FORMAT
+        };
+        final Object[] vals = new Object[]{expectedCopyNumber, copyNumber};
+        return SVTestUtils.keyValueArraysToMap(keys, vals);
+    }
+
+    private Map<String, Object> createGenotypeTestAttributes(final Integer expectedCopyNumber) {
+        return Collections.singletonMap(GATKSVVCFConstants.EXPECTED_COPY_NUMBER_FORMAT, expectedCopyNumber);
+    }
+
+    @DataProvider(name = "collapseSampleGenotypesTestData")
+    public Object[][] collapseSampleGenotypesTestData() {
```

Comment:
These tests are very nice. I worry a little that they are somewhat hard to find initially for someone doing a bug fix -- say you wanted to fix a problem in the method `CanonicalSVCollapser.getCNVGenotypeAllelesFromCopyNumber()` -- you'd have to know to come to `SVCollapserTest.collapseSampleGenotypesTestData()` to find your test case. This class only seems to be testing the `CanonicalSVCollapser` -- what about renaming it to `CanonicalSVCollapserUnitTest`?

Reply:
Good suggestion (I think I forgot to change this after refactoring)